Sources/IRGen/Expression/IRBinaryExpression.swift

```diff
@@ -30,26 +30,35 @@ struct IRBinaryExpression {
     let rhs = IRExpression(expression: binaryExpression.rhs, asLValue: asLValue)
       .rendered(functionContext: functionContext)
 
+    var preamble = lhs.preamble + "\n" + rhs.preamble
+    let lhsExp = lhs.expression
+    let rhsExp = rhs.expression
+    var code = ""
+
     switch binaryExpression.opToken {
     case .equal:
-      return IRAssignment(lhs: binaryExpression.lhs, rhs: binaryExpression.rhs)
+      let assign = IRAssignment(lhs: binaryExpression.lhs, rhs: binaryExpression.rhs)
         .rendered(functionContext: functionContext)
-
-    case .plus: return IRRuntimeFunction.add(a: lhs, b: rhs)
-    case .overflowingPlus: return "add(\(lhs), \(rhs))"
-    case .minus: return IRRuntimeFunction.sub(a: lhs, b: rhs)
-    case .overflowingMinus: return "sub(\(lhs), \(rhs))"
-    case .times: return IRRuntimeFunction.mul(a: lhs, b: rhs)
-    case .overflowingTimes: return "mul(\(lhs), \(rhs))"
-    case .divide: return IRRuntimeFunction.div(a: lhs, b: rhs)
-    case .closeAngledBracket: return "gt(\(lhs), \(rhs))"
-    case .openAngledBracket: return "lt(\(lhs), \(rhs))"
-    case .doubleEqual: return "eq(\(lhs), \(rhs))"
-    case .notEqual: return "iszero(eq(\(lhs), \(rhs)))"
-    case .or: return "or(\(lhs), \(rhs))"
-    case .and: return "and(\(lhs), \(rhs))"
-    case .power: return IRRuntimeFunction.power(b: lhs, e: rhs)
+      preamble += "\n" + assign.preamble
+      code = assign.expression
+    case .plus:
```

Comment:
Would it perhaps be cleaner if we extract this whole switch statement into a separate function? Then we can save ourselves from assigning to `code` a bunch.